### ISSUE TITLE
ERROR: keystone authentication failed when install opensds hotpot

### ISSUE THREAD

**Issue by CongMinYin** [opensds/opensds#1031]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
> /kind bug

**What happened**:
ERROR: keystone authentication failed

**What you expected to happen**:
It should installed successful.

**How to reproduce it (as minimally and precisely as possible)**:
I follow the install guide.
Firstly I read https://github.com/opensds/opensds/wiki/OpenSDS-Integration-with-Kubernetes-CSI
I installed gcc make libc6-dev, go and docker.
Next, I read https://github.com/opensds/opensds/wiki/OpenSDS-Local-Cluster-with-Multi-tenants-Installation
I installed git, make and gcc.
I execute the command   curl -sSL https://raw.githubusercontent.com/opensds/opensds/master/install/devsds/bootstrap.sh | sudo bash
when it successful, I modify install/dev/local.conf, change noauth to keystone.
Finally, I run ./install/dev/install.sh but ERROR occur, keystone authentication failed.

My environment is Ubuntu 18.04, kernel version is 5.0.0. I tried several times in clean ubuntu system in virtual box.You can see more details in the following picture.

![Screenshot from 2019-10-23 15-03-53](https://user-images.githubusercontent.com/13258081/67367499-368c8380-f5a8-11e9-94c2-6d48fb8f5fe4.png)

**Anything else we need to know?**:

**Environment**:
- Hotpot(release/branch) version: It's from curl ....install/devsds/bootstrap.sh | sudo bash
- OS (e.g. from /etc/os-release):18.04.3 LTS (Bionic Beaver)
- Kernel (e.g. `uname -a`):5.0.0-23-generic
- Install tools:install/dev/install.sh
- Others:

**Comment by joseph-v** [comment]
Looks like osdsapiserver is not running. Can you please check if etcd is running? Also, can you try the steps in Ubuntu 16.04?

**Comment by CongMinYin** [comment]
Yes, osdsapiserver is not running,etcd is running. I deployed opensds environment more than 10 times in 18.04 in different machines. Some are ok, others failed in keystone failed. I don't know the reason or the difference between those machines.
![image](https://user-images.githubusercontent.com/13258081/68176020-3172ff80-ffbe-11e9-9e23-dbe74768c3d9.png)

**Comment by joseph-v** [comment]
Can you please check, if any info for failure in osdsapiserver log file (/var/log/opensds/osdsapiserver.INFO) ?

**Comment by CongMinYin** [comment]
Hi, this file show:
root@develop:~/gopath/src/github.com/opensds/opensds# cat /var/log/opensds/osdsapiserver.INFO
Log file created at: 2019/11/07 10:48:30
Running on machine: develop
Binary: Built with gc go1.12.1 for linux/amd64
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
I1107 10:48:30.426708   19078 logs.go:75] [Info] LogFlushFrequency: 5s
I1107 10:48:30.427568   19078 auth.go:48] keystone
E1107 10:48:30.480369   19078 keystone.go:73] When get auth client:Internal Server Error
root@develop:~/gopath/src/github.com/opensds/opensds#

**Comment by Wangzihao18** [comment]
Same error on CentOS 7.6, but the osdsapiserver is runnung, osdsdock is not running
![image](https://user-images.githubusercontent.com/55722482/70106552-ec0a2680-167e-11ea-9eda-a22d086e616a.png)

**Comment by wisererik** [comment]
from your log, i guess it is os compatibility problem, we just tested opensds on ubuntu 16.04 as of now. 
ubuntu 18.04 and centos is not tested, i was hoping that you could help us fix these issues. @joseph-v can help you if you have more information, such as /var/log/opensds/osdsdock.ERROR

**Comment by Wangzihao18** [comment]
Hi, thanks @wisererik . I will try to run opensds on ubuntu16.04 first to get familiar with opensds. Then try to fix the error on ubuntu18 or centos. And there is no `/var/log/opensds` dictionary on centos.

**Comment by wisererik** [comment]
I just followed the [wiki](https://github.com/opensds/opensds/wiki/OpenSDS-Cluster-Installation-through-Ansible) to install opensds on ubuntu 1804, it works fine. 
```
TASK [orchestration-installer : Start orchestration manager] *******************
skipping: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=62   changed=38   unreachable=0    failed=0   

root@osboxes:~/opensds-installer/ansible# cat /etc/issue
Ubuntu 18.04.3 LTS \n \l

root@osboxes:~/opensds-installer/ansible# vgs
  VG              #PV #LV #SN Attr   VSize   VFree  
  opensds-nfs       1   0   0 wz--n- <10.00g <10.00g
  opensds-volumes   1   0   0 wz--n- <10.00g <10.00g
root@osboxes:~/opensds-installer/ansible# ps -ef | grep osds
root     20701  1464  0 09:44 pts/0    00:00:00 grep --color=auto osds
root     30623     1  0 09:06 pts/0    00:00:00 bin/osdslet
root     30666     1  0 09:06 pts/0    00:00:00 bin/osdsapiserver
root     31437     1  0 09:07 pts/0    00:00:00 bin/osdsdock

```

this is my ansible configuration in ansible/group_vars/common.yml 
```
# This field indicates local machine host ip
host_ip: 192.168.56.31

# This field indicates which project should be deploy
# 'hotpot', 'gelato' or 'all'
deploy_project: all

# This field indicates which way user prefers to install, currently support
# 'repository', 'release' and 'container'
install_from: release
# These fields below will specify the tag based on install_from type
repo_branch: master
release_version: v0.6.3
image_tag: latest

```

**Comment by Wangzihao18** [comment]
> I just followed the [wiki](https://github.com/opensds/opensds/wiki/OpenSDS-Cluster-Installation-through-Ansible) to install opensds on ubuntu 1804, it works fine.
> 
> ```
> TASK [orchestration-installer : Start orchestration manager] *******************
> skipping: [localhost]
> 
> PLAY RECAP *********************************************************************
> localhost                  : ok=62   changed=38   unreachable=0    failed=0   
> 
> root@osboxes:~/opensds-installer/ansible# cat /etc/issue
> Ubuntu 18.04.3 LTS \n \l
> 
> root@osboxes:~/opensds-installer/ansible# vgs
>   VG              #PV #LV #SN Attr   VSize   VFree  
>   opensds-nfs       1   0   0 wz--n- <10.00g <10.00g
>   opensds-volumes   1   0   0 wz--n- <10.00g <10.00g
> root@osboxes:~/opensds-installer/ansible# ps -ef | grep osds
> root     20701  1464  0 09:44 pts/0    00:00:00 grep --color=auto osds
> root     30623     1  0 09:06 pts/0    00:00:00 bin/osdslet
> root     30666     1  0 09:06 pts/0    00:00:00 bin/osdsapiserver
> root     31437     1  0 09:07 pts/0    00:00:00 bin/osdsdock
> ```
> 
> this is my ansible configuration in ansible/group_vars/common.yml
> 
> ```
> # This field indicates local machine host ip
> host_ip: 192.168.56.31
> 
> # This field indicates which project should be deploy
> # 'hotpot', 'gelato' or 'all'
> deploy_project: all
> 
> # This field indicates which way user prefers to install, currently support
> # 'repository', 'release' and 'container'
> install_from: release
> # These fields below will specify the tag based on install_from type
> repo_branch: master
> release_version: v0.6.3
> image_tag: latest
> ```

ok ! thanks very much . I will try it ^_^

**Comment by CongMinYin** [comment]
Hi Erik,

I installed in devsds way in ubuntu 18.04, not ansible,  refer to:
https://github.com/opensds/opensds/wiki/OpenSDS-Integration-with-Kubernetes-CSI
https://github.com/opensds/opensds/wiki/OpenSDS-Local-Cluster-with-Multi-tenants-Installation
I tried more than 10 times to deployed opensds. It works fine in some of my machine. And sometimes, it failed. The environments are the same. I don’t know why. Now, I have to use noauth to install OpenSDS Because of the possibility of keystone failure.

Best regards.
Congmin Yin

**Comment by wisererik** [comment]
Hi @CongMinYin I will try devsds in my ubuntu1804
